numpy/linalg/tests/test_linalg.py

```diff
@@ -2372,6 +2372,22 @@ def test_matrix_norm():
     assert_almost_equal(actual, np.array([[14.2828]]), double_decimal=3)
 
 
+def test_matrix_norm_empty():
+    for shape in [(0, 2), (2, 0), (0, 0)]:
+        x = np.zeros(shape)
```

Comment:
This only tests for type `float64`. Could you also add a case for a few other types? (not all types are needed, maybe just int and float32)